### ISSUE TITLE
large CER

### ISSUE THREAD

**Issue by ZCzzzzzz** [speechbrain/speechbrain#2669]
### Describe the bug

Hello, because I want to reason, I commented out the Training part in the train.py, only the test part, and found that I can reason, but the CER I get is very large, what should I do?
![large CER](https://github.com/user-attachments/assets/2056d1a5-7d5f-418e-b6f1-54302e976ba3)

### Expected behaviour

Also, can you tell me how CER is calculated in this project? Where is the code for the CER calculation?

**Comment by TParcollet** [comment]
Hi, I am not sure about what you are trying to achieve. You can perform only testing with the --test_only flag, no need for commenting. The code for CER calculation is, as visible in the yaml file, here: speechbrain.utils.metric_stats.ErrorRateStats

**Comment by ZCzzzzzz** [comment]
Sorry, I don't quite understand what you mean, can you be specific? I typed in the "python train.py hparams/train_ASR_transformer.yaml --data_folder=/localscratch/aishell/" command before and it did work with training, and the test_cer I got in the end was normal, and now I want to inference, and I see that train.py has "Testing" behind it, so I put the previous " Training" part is commented out, and the program can also run, but the CER value is obviously abnormal，what should I do?

**Comment by TParcollet** [comment]
Right, first thing to do then, have a look at our [inference tutorial](https://speechbrain.github.io/tutorial_advanced.html] that will give you more information about what you are doing and trying to do. For the flag, writing: python train.py hparams/train_ASR_transformer.yaml --data_folder=/localscratch/aishell/ --test_only should skip the training. If the training has been done fully, then the test should just work (if the checkpoints are loaded, this should be logged in the terminal).

**Comment by ZCzzzzzz** [comment]
Hello, I reasoned as you said using "python train.py hparams/train_ASR_transformer.yaml --data_folder=/localscratch/aishell/ --test_only", but ran into "Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/speechbrain/utils/checkpoints.py", line 1165, in average_state_dicts
    running_sum = next(iterator)
StopIteration", how do I fix it?

**Comment by TParcollet** [comment]
Well it looks like your model hasn't been trained (i.e. no checkpoints to load)? Could you look into the results folder? There should be a save folder with CKPT-{smth} in it. If not, the model hasn't been trained. Also, I need to know what recipe does that. Is this a standard SB recipe? You need to let the training happen for at least one epoch to enable the test like this, because otherwise no checkpoints exist.

**Comment by ys830** [comment]
May I ask what the final test results were when you ran it on the AISHELL dataset? I used the speechbrain/asr-wav2vec2-ctc-aishell model and ran the --test_only command, and the results I got were:
```
speechbrain.core - Beginning experiment!
speechbrain.core - Experiment folder: results/ctc_wav2vec2/2
speechbrain.core - Info: precision arg from hparam file is used
speechbrain.core - Info: ckpt_interval_minutes arg from hparam file is used
speechbrain.core - Gradscaler enabled: False. Using precision: fp32.
speechbrain.core - ASR Model Statistics:
* Total Number of Trainable Parameters: 340.2M
* Total Number of Parameters: 340.2M
* Trainable Parameters represent 100.0000% of the total size.
speechbrain.decoders.ctc - space_token ` ` not found in the vocabulary.Using value -1 as `space_index`.Note: If your transcription is not expected to contain spaces, you can ignore this warning.
speechbrain.decoders.ctc - Found `space_token` at index -1.
speechbrain.core - Test only mode, skipping training and validation stages.
speechbrain.utils.checkpoints - Would load a checkpoint here, but none found yet.
torch.distributed.nn.jit.instantiator - Created a temporary directory at /tmp/tmp6l2e2f1l                                                                                                                                         
torch.distributed.nn.jit.instantiator - Writing /tmp/tmp6l2e2f1l/_remote_module_non_scriptable.py                                                                                                                                 
speechbrain.utils.train_logger - Epoch loaded: 0 - test loss: 1.44e+02, test CER: 1.33e+03
```
These results differ somewhat from those provided by the official model. Do you know what might be the reason for this?
![image](https://github.com/user-attachments/assets/7157741a-3706-441a-8ee7-2a0c23876e84)
